Mailnag/backends/local.py

```diff
@@ -82,3 +82,66 @@ def notify_next_change(self, callback=None, timeout=None):
 	def cancel_notifications(self):
 		raise NotImplementedError("mbox does not support notifications")
 
+
+class MaildirBackend(MailboxBackend):
+	"""Implementation of maildir mail boxes."""
+
+	def __init__(self, name = '', path=None, **kw):
+		"""Initialize maildir mailbox backend with a name and path."""
+		self.name = name
+		self.path = path
+		self.opened = False
+
+
+	def open(self, reopen=False):
+		"""'Open' mailbox. (Actually just checks that maildir directory exists.)"""
+		if not os.path.isdir(self.path):
+			raise IOError('Mailbox {} does not exist.'.format(self.path))
+		self.opened = True
+
+
+	def close(self):
+		"""Close mailbox."""
+		self.opened = False
+
+
+	def is_open(self):
+		"""Return True if mailbox is opened."""
+		return self.opened
+
+
+	def list_messages(self):
+		"""List unread messages from the mailbox.
+		Yields pairs (folder, message) where folder is always ''.
+		"""
+		maildir = mailbox.Maildir(self.path, factory=None, create=False)
+		folder = ''
+		try:
+			for msg in maildir:
+				if 'S' not in msg.get_flags():
+					yield folder, msg
+		finally:
+			maildir.close()
+
+
+	def request_folders(self):
+		"""Lists folder from maildir recursively."""
+
+		def list_folders(maildir, parent):
+			for folder in  maildir.list_folders():
+				this_folder = parent + folder
+				yield this_folder
+				for subfolder in list_folders(maildir.get_folder(folder), this_folder + '/'):
+					yield subfolder
+
+		maildir = mailbox.Maildir(self.path, factory=None, create=False)
+		return [''] + list(list_folders(maildir, ''))
```

Comment:
It is the root folder. list_folders function lists only folders below that.
In the account dialog that should be expressed some other way thought.